main/test/flix/Test.Exp.Fixpoint.Permutations.flix

```diff
@@ -0,0 +1,863 @@
+/*
+ * Copyright 2025 Casper Dalgaard Nielsen
+ *                Adam Yasser Tallouzi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+mod Test.Exp.Fixpoint.Permutations {
+
+    /////////////////////////////////////////////////////////////////////////////
+    // Test permuting body atom order with Int32                               //
+    /////////////////////////////////////////////////////////////////////////////
+
+    // May not be bigger than 12.
+    def smallN(): Int32 = 10
+
+    def resultForFirstInt32(n: Int32): Vector[(Int32, Int32, Int32, Int32)] = {
+        Vector.range(0, n - 2) |> Vector.map(x -> (x, x + 1, x + 2, x + 3))
+    }
+
+    @Test
+    def testPermutation01(): Bool = {
+        let n = smallN();
+        let a_ = inject Vector.range(0, n) |> Vector.map(x -> (x, x + 1)) into A/2;
```

Comment:
I would avoid variables that end with underscores. To easy to confused with variables that start with underscore (i.e. unused variables)